service/storage_proxy.cc

```diff
@@ -1488,33 +1501,16 @@ class write_response_handler : public abstract_write_response_handler {
             db::consistency_level cl, db::write_type type,
             std::unique_ptr<mutation_holder> mh, inet_address_vector_replica_set targets,
             const inet_address_vector_topology_change& pending_endpoints, inet_address_vector_topology_change dead_endpoints, tracing::trace_state_ptr tr_state,
-            storage_proxy::write_stats& stats, service_permit permit, db::per_partition_rate_limit::info rate_limit_info) :
+            storage_proxy::write_stats& stats, service_permit permit, db::per_partition_rate_limit::info rate_limit_info, is_cancellable cancellable) :
                 abstract_write_response_handler(std::move(p), std::move(ermp), cl, type, std::move(mh),
-                        std::move(targets), std::move(tr_state), stats, std::move(permit), rate_limit_info, pending_endpoints.size(), std::move(dead_endpoints)) {
+                        std::move(targets), std::move(tr_state), stats, std::move(permit), rate_limit_info, pending_endpoints.size(), std::move(dead_endpoints), cancellable) {
         _total_endpoints = _targets.size();
     }
 };
 
-class view_update_write_response_handler : public write_response_handler, public bi::list_base_hook<bi::link_mode<bi::auto_unlink>> {
-public:
-    view_update_write_response_handler(shared_ptr<storage_proxy> p,
-            locator::effective_replication_map_ptr ermp,
-            db::consistency_level cl,
-            std::unique_ptr<mutation_holder> mh, inet_address_vector_replica_set targets,
-            const inet_address_vector_topology_change& pending_endpoints, inet_address_vector_topology_change dead_endpoints, tracing::trace_state_ptr tr_state,
-            storage_proxy::write_stats& stats, service_permit permit, db::per_partition_rate_limit::info rate_limit_info):
-                write_response_handler(p, std::move(ermp), cl, db::write_type::VIEW, std::move(mh),
-                        std::move(targets), pending_endpoints, std::move(dead_endpoints), std::move(tr_state), stats, std::move(permit), rate_limit_info) {
-        register_in_intrusive_list(*p);
-    }
-    ~view_update_write_response_handler();
-private:
-    void register_in_intrusive_list(storage_proxy& p);
-};
-
```

Comment:
@gleb-cloudius we should consolidate all the *write_response_handler:s into a single class with no virtual functions. They're just adding complexity.

Reply:
They encapsulate different ways in which different CLs are handled. Differences are no huge indeed, but still will have to be replaced by switches on a CL type.

Reply:
Yes, but we can abstract it on a in a different that has less repetition.